src/sentry/static/sentry/app/utils/recreateRoute.tsx

```diff
@@ -0,0 +1,55 @@
+import {Location} from 'history';
+import {PlainRoute} from 'react-router';
+import {findLastIndex} from 'lodash';
+
+import replaceRouterParams from 'app/utils/replaceRouterParams';
+
+type Options = {
+  routes: PlainRoute[];
+  location: Location;
+
+  // parameters to replace any route string parameters (e.g. if route is `:orgId`,
+  // params should have `{orgId: slug}`
+  params: {[key: string]: string};
+
+  /**
+   * The number of routes to to pop off of `routes
+   * Must be < 0
+   *
+   * There's no ts type for negative numbers so we are arbitrarily specifying -1-9
+   */
+  stepBack?: -1 | -2 | -3 | -4 | -5 | -6 | -7 | -8 | -9;
+};
+
+/**
+ * Given a route object or a string and a list of routes + params from router, this will attempt to recreate a location string while replacing url params.
+ * Can additionally specify the number of routes to move back
+ *
+ * See tests for examples
+ */
+export default function recreateRoute(to: string | PlainRoute, options: Options): string {
+  const {routes, params, location, stepBack} = options;
+  const paths = routes.map(({path}) => path || '');
+  let lastRootIndex: number;
+  let routeIndex: number | undefined;
+
+  // TODO(ts): typescript things
```

Comment:
Previously this was assigned to a local var, but does not pass typescript type checking (bug in ts).